### ISSUE TITLE
How to run inference

### ISSUE THREAD

**Issue by bamba518** [ARM-software/ML-KWS-for-MCU#42]
Not sure if this is the right place, but if someone can point me on how to do live microphone inference on the FRDM-K64F, I built the .bin file and loaded it, but not sure what to do next

**Comment by leejaeuk** [comment]
You want to real-time test in FRDM-K64F?
i successfully operation my FRDM-K64F. it maybe helpful to you.

`#include "kws_ds_cnn.h"
#include "mbed.h"
//#include <intrin.h>
//#include "wav_data.h"

AnalogIn soundin(A0);
Serial pc(USBTX, USBRX); // tx, rx
Ticker sample;
void sample_sound(void);
volatile short int sound;

int i=0;
int j=0;
int count =0;
int z=0;

float s;
uint16_t u16_sound;
uint8_t sendl,sendu;
int8_t buffer_l[5000]; 
int8_t buffer_u[5000];
int16_t audio_input_buffer[16000];

void sample_sound(void)
{   

    while(i <16000){
        u16_sound =((soundin.read_u16()));
        unsigned short _byteswap_ushort(u16_sound);
        audio_input_buffer[i] = ((int16_t)(_byteswap_ushort));
        //pc.printf("%d  ", audio_input_buffer[i]);
        wait(0.0000625f);
        i++;
    }

    if(i > 15999) {
        sample.detach();
        DigitalOut led(LED1);
        //ready=true;
        sample.detach();
    }    
}


Timer T;


int main()
{
  pc.baud(9600);
  sample.attach_us(sample_sound, 125);
  int j=0;

  char output_class[12][8] = {"Silence", "Unknown","yes","no","up","down","left","right","on","off","stop","go"};
  KWS_DS_CNN kws(audio_input_buffer);

  T.start();
  int start=T.read_us();
  kws.extract_features(); //extract mfcc features
  kws.classify();	  //classify using dnn
  int end=T.read_us();
  T.stop();
  int max_ind = kws.get_top_class(kws.output);
  pc.printf("Total time : %d us\r\n",end-start);
  printf("Detected %s (%d%%)\r\n",output_class[max_ind],((int)kws.output[max_ind]*100/128));

  return 0;
}`
